mypy/fixup.py

```diff
@@ -0,0 +1,298 @@
+"""Fix up various things after deserialization."""
+
+from typing import Any, Dict, Optional, cast
+
+from mypy.nodes import (MypyFile, SymbolNode, SymbolTable, SymbolTableNode,
+                        TypeInfo, FuncDef, OverloadedFuncDef, Decorator, Var,
+                        TypeVarExpr, ClassDef,
+                        LDEF, MDEF, GDEF, MODULE_REF)
+from mypy.types import (CallableType, EllipsisType, Instance, Overloaded, TupleType,
+                        TypeList, TypeVarType, UnboundType, UnionType, TypeVisitor)
+from mypy.visitor import NodeVisitor
+
+
+def fixup_module_pass_one(tree: MypyFile, modules: Dict[str, MypyFile]) -> None:
+    node_fixer = NodeFixer(modules)
+    node_fixer.visit_symbol_table(tree.names)
+    # print('Done pass 1', tree.fullname())
+
+
+def fixup_module_pass_two(tree: MypyFile, modules: Dict[str, MypyFile]) -> None:
+    compute_all_mros(tree.names, modules)
+    # print('Done pass 2', tree.fullname())
+
+
+def compute_all_mros(symtab: SymbolTable, modules: Dict[str, MypyFile]) -> None:
+    for key, value in symtab.items():
+        if value.kind in (LDEF, MDEF, GDEF) and isinstance(value.node, TypeInfo):
+            info = value.node
+            # print('  Calc MRO for', info.fullname())
+            try:
+                info.calculate_mro()
+            except Exception:
+                import pdb
```

Comment:
Hmm maybe this shouldn't be the default behavior? If you want to keep this, add a TODO comment about removing this before turning this on by default.